components/scream/src/share/util/scream_utils.hpp

```diff
@@ -129,6 +129,29 @@ void transpose(const Scalar* sv, Scalar* dv, Int ni, Int nk) {
         dv[nk*i + k] = sv[ni*k + i];
 };
 
+// A substitute for std::pow when exp is a positive integer. This
+// implementation is ~2x faster than std::pow, but, more importantly,
+// it is BFB with Fortran's "**" operator when base is a floating point
+// (std::pow is not!).
+template <typename Scalar> KOKKOS_INLINE_FUNCTION
```

Comment:
This is nice, but we should avoid using it. Instead, implement `square` and `cube` routines using `x*x` and `x*x*x`, which I think will be all we need most of the time. I believe C++ `cube(x)` will then be BFB with F90 `x**3`.

Reply:
You're exactly right. I found out later that this approach was still non-bfb and the only way to guarantee bfb was to unroll the pow into multiplication as you said. Should I just delete this routine or do you think it may have value if we ever encounter a larger integer exponent?

Reply:
IMO, delete it; it has the strong potential to be misused, as pow often is.